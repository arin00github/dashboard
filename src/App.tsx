import React, { useState } from "react";

import {
    createBrowserRouter,
    createRoutesFromElements,
    Route,
    RouterProvider,
    createMemoryRouter,
    MemoryRouter,
    Routes,
} from "react-router-dom";

import { BasicMenu } from "./views/layouts/menuRouter";
import { OpenLayout } from "./views/layouts/OpenLayout";
import { ProtectedLayout } from "./views/layouts/ProtectedLayout";
import LoginPage from "./views/pages/login/LoginPage";

export const NoMatch = () => {
    return <h3>no match</h3>;
};

export const routerFrame = createRoutesFromElements(
    <>
        <Route path="/" element={<ProtectedLayout />}>
            {BasicMenu.map((menu) => (
                <Route path={menu.href} element={menu.component} key={`router_menu_${menu.title}`} />
            ))}
            <Route path="*" element={<NoMatch />} />
        </Route>
        <Route path="/" element={<OpenLayout />}>
            <Route path="/login" element={<LoginPage />} />
            <Route path="*" element={<NoMatch />} />
        </Route>
    </>
);

export const RouterContainer = () => {
    return (
        <Routes>
            <Route path="/" element={<ProtectedLayout />}>
                {BasicMenu.map((menu) => (
                    <Route path={menu.href} element={menu.component} key={`router_menu_${menu.title}`} />
                ))}
                <Route path="*" element={<NoMatch />} />
            </Route>
            <Route path="/" element={<OpenLayout />}>
                <Route path="/login" element={<LoginPage />} />
                <Route path="*" element={<NoMatch />} />
            </Route>
        </Routes>
    );
};

const router =
    process.env.NODE_ENV === "test"
        ? createMemoryRouter(routerFrame, { initialEntries: ["/", "/login"] })
        : createBrowserRouter(routerFrame);

function App() {
    return <RouterProvider router={router} />;
}

export default App;
