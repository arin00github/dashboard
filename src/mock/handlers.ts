import { rest } from "msw";

export function makeUrl(path: string) {
    return `https://localhost:8080${path}`;
}

const postData = [
    {
        userId: 1,
        id: 1,
        title: "sunt aut facere repellat provident occaecati excepturi optio reprehenderit",
        body: "aaaa",
    },
    { userId: 1, id: 2, title: "qui est esse", body: "aaaa" },
    { userId: 1, id: 3, title: "ea molestias quasi exercitationem repellat qui ipsa sit aut", body: "aaaa" },
];

export const handlers = [
    rest.post(makeUrl("/login"), (req, res, ctx) => {
        // Persist user's authentication in the session
        //sessionStorage.setItem("is-authenticated", "true");
        return res(
            // Respond with a 200 status code
            ctx.status(200)
        );
    }),
    rest.get("/user", (req, res, ctx) => {
        // Check if the user is authenticated in this session
        // const isAuthenticated = sessionStorage.getItem("is-authenticated");
        // if (!isAuthenticated) {
        //     // If not authenticated, respond with a 403 error
        //     return res(
        //         ctx.status(403),
        //         ctx.json({
        //             errorMessage: "Not authorized",
        //         })
        //     );
        // }
        // If authenticated, return a mocked user details
        return res(
            ctx.status(200),
            ctx.json({
                username: "admin",
            })
        );
    }),

    // rest.get("https://jsonplaceholder.typicode.com/posts", (req, res, ctx) => {
    //     return res(ctx.status(200), ctx.json(postData), ctx.delay(150));
    // }),
];
