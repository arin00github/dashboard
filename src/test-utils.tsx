import React from "react";

import { configureStore } from "@reduxjs/toolkit";
import { render, RenderOptions } from "@testing-library/react";
import { Provider } from "react-redux";
import { PreloadedState } from "redux";
import { Persistor } from "redux-persist";
import { PersistGate } from "redux-persist/integration/react";

import { AppStore, persistedReducer, RootState, persistor as PersistorSelf } from "./redux/store";

interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
    preloadedState?: PreloadedState<RootState>;
    store?: AppStore;
    persistor?: Persistor;
}

export function renderWithProviders(
    ui: React.ReactElement,
    {
        preloadedState = {},
        store = configureStore({ reducer: persistedReducer, preloadedState }),
        persistor = PersistorSelf,
    }: ExtendedRenderOptions = {}
) {
    function Wrapper({ children }: { children: React.ReactNode }): JSX.Element {
        return (
            <PersistGate persistor={persistor}>
                <Provider store={store}>{children}</Provider>
            </PersistGate>
        );
    }
    return { store, ...render(ui, { wrapper: Wrapper }) };
}

export * from "@testing-library/react";
