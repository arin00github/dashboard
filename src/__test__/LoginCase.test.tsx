import React from "react";

import { rest } from "msw";
import { createMemoryRouter, MemoryRouter, RouterProvider } from "react-router-dom";

import App, { RouterContainer, routerFrame } from "../App";
import { server } from "../mock/server";
import { cleanup, fireEvent, renderWithProvidersNoRouter, screen, waitFor } from "../test-utils";

afterEach(cleanup);

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
    ...jest.requireActual("react-router-dom"),
    useNavigate: () => mockedUsedNavigate,
}));

describe("Login Sinario", () => {
    it("0. Initiat Project", () => {
        const renderer = renderWithProvidersNoRouter(
            <MemoryRouter initialEntries={["/login"]}>
                <RouterContainer />
            </MemoryRouter>
        );

        expect(renderer).toMatchSnapshot();
    });

    // it("1. Display login Page", async () => {
    //     console.log("location", window.location.pathname);
    //     if (window.location.pathname === "/login") {
    //         expect(screen.getByText("Login")).toBeInTheDocument();
    //     } else {
    //         console.log("login", screen.getByText("Login"));
    //         //expect(mockedUsedNavigate).toHaveBeenCalledTimes(1);
    //     }
    // });
});
