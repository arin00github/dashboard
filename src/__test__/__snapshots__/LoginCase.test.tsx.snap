// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Sinario 0. Initiat Project 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gKPRtg hALsVB"
      >
        <div>
          <div
            class="sc-eDvSVe dsxVKz"
          >
            <h2
              aria-label="page-title-Login"
            >
              Login
            </h2>
          </div>
          <form
            action=""
          >
            <div
              class="mb-4 col-sm-12"
            >
              <label
                class="sc-bcXHqe eaQZiJ"
                for="basic-userid"
              >
                userid
              </label>
              <input
                aria-label="basic-userid"
                class="input-box"
                id="basic-userid"
                name="userid"
                placeholder="userid"
                role="textbox"
                type="text"
                value=""
              />
            </div>
            <div
              class="mb-4 col-sm-12"
            >
              <label
                class="sc-bcXHqe eaQZiJ"
                for="basic-email"
              >
                email
              </label>
              <input
                aria-label="basic-email"
                class="input-box"
                id="basic-email"
                name="email"
                placeholder="email"
                role="textbox"
                type="text"
                value=""
              />
            </div>
            <small
              aria-label="error-box"
              class="form-text"
            />
            <button
              aria-label="submit-btn"
              class="sc-hLBbgP gVFxkb btn w-100"
              role="button"
            >
              submit
            </button>
          </form>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gKPRtg hALsVB"
    >
      <div>
        <div
          class="sc-eDvSVe dsxVKz"
        >
          <h2
            aria-label="page-title-Login"
          >
            Login
          </h2>
        </div>
        <form
          action=""
        >
          <div
            class="mb-4 col-sm-12"
          >
            <label
              class="sc-bcXHqe eaQZiJ"
              for="basic-userid"
            >
              userid
            </label>
            <input
              aria-label="basic-userid"
              class="input-box"
              id="basic-userid"
              name="userid"
              placeholder="userid"
              role="textbox"
              type="text"
              value=""
            />
          </div>
          <div
            class="mb-4 col-sm-12"
          >
            <label
              class="sc-bcXHqe eaQZiJ"
              for="basic-email"
            >
              email
            </label>
            <input
              aria-label="basic-email"
              class="input-box"
              id="basic-email"
              name="email"
              placeholder="email"
              role="textbox"
              type="text"
              value=""
            />
          </div>
          <small
            aria-label="error-box"
            class="form-text"
          />
          <button
            aria-label="submit-btn"
            class="sc-hLBbgP gVFxkb btn w-100"
            role="button"
          >
            submit
          </button>
        </form>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
